server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingCache.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.cluster.ClusterChangedEvent;
+import org.opensearch.cluster.ClusterStateListener;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.cache.Cache;
+import org.opensearch.common.cache.CacheBuilder;
+import org.opensearch.common.lease.Releasable;
+import org.opensearch.index.shard.ShardId;
+
+import java.util.List;
+
+/**
+ * The weighted shard routing cache allows caching shard routing iterator returned by Weighted round-robin scheduling
+ * policy, helping with improving similar requests.
+ *
+ * @opensearch.internal
+ */
+public class WeightedRoutingCache implements Releasable, ClusterStateListener {
```

Comment:
Can we drop the Cache based on [comment](https://github.com/opensearch-project/OpenSearch/pull/4241/files#r971681349)

Reply:
Removed WeightedRoutingCache and created a map in IndexShardRoutingTable to store routings in memory until weights or shard routing table changes.